opentelemetry-sdk/benches/batch_log_processor.rs

```diff
@@ -0,0 +1,95 @@
+//! run with `$ cargo bench --bench batch_log_processor --features="rt-tokio,testing"
+//!
+/*
+The benchmark results:
+criterion = "0.5.1"
+OS: Ubuntu 22.04.3 LTS (6.6.36.3-microsoft-standard-WSL2)
+Hardware: AMD EPYC 7763 64-Core Processor - 2.44 GHz, 16vCPUs,
+RAM: 32.0 GB
+| Test                           | Average time|
+|--------------------------------|-------------|
+| batch_log_processor_emit       |  438.47 ns  |
+*/
+use criterion::{criterion_group, criterion_main, Criterion};
+use opentelemetry::logs::LoggerProvider as _;
+use opentelemetry::logs::Severity;
+use opentelemetry::logs::{LogRecord, Logger};
+use opentelemetry_sdk::export::logs::LogExporter;
+use opentelemetry_sdk::logs::LoggerProvider;
+use opentelemetry_sdk::logs::{BatchConfigBuilder, BatchLogProcessor};
+use opentelemetry_sdk::resource::Resource;
+use opentelemetry_sdk::runtime::Tokio;
+use std::time::Instant;
+use std::time::SystemTime;
+use tokio::runtime::Runtime;
+
+#[derive(Debug)]
+struct NoopLogExporter;
+
+#[async_trait::async_trait]
+impl LogExporter for NoopLogExporter {
+    async fn export<'a>(
+        &mut self,
+        _batch: Vec<std::borrow::Cow<'a, opentelemetry_sdk::export::logs::LogData>>,
+    ) -> opentelemetry::logs::LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&mut self) {}
+
+    fn set_resource(&mut self, _resource: &Resource) {}
+}
+
+fn benchmark_batch_log_processor(c: &mut Criterion) {
+    // Benchmark 1: Create record, and emit it.
+    let rt = Runtime::new().unwrap();
+    let max_queue_size = 10_000;
+    c.bench_function("batch_log_processor_emit", |b| {
+        rt.block_on(async {
+            let exporter = NoopLogExporter;
+            // Create the BatchConfig with the desired queue and batch size
+            let batch_config = BatchConfigBuilder::default()
+                .with_max_queue_size(10_000)
+                .with_scheduled_delay(std::time::Duration::from_secs(1))
+                .build();
+
+            let log_processor = BatchLogProcessor::builder(exporter, Tokio)
+                .with_batch_config(batch_config)
+                .build();
+            let provider = LoggerProvider::builder()
+                .with_log_processor(log_processor)
+                .build();
+            let logger = provider.logger("test-logger");
+
+            b.iter_custom(|iters| {
+                let mut count = 0;
+                let mut total_duration = std::time::Duration::new(0, 0);
+                for _ in 0..iters {
+                    if count == max_queue_size - 1 {
+                        // flush the queue outside the timing loop
+                        provider.force_flush();
```

Comment:
what is the need for this flush?

Reply:
The queue gets full rapidly without this, even with a 1sec upload interval.